go/engine/device_add.go

```diff
@@ -100,7 +99,8 @@ func (e *DeviceAdd) Run(ctx *Context) (err error) {
 			e.G().Log.Debug("received secret phrase, adding to provisioner")
 			ks, err := libkb.NewKex2SecretFromPhrase(receivedSecret.Phrase)
 			if err != nil {
-				e.G().Log.Warning("DisplayAndPromptSecret error: %s", err)
+				e.G().Log.Warning("NewKex2SecretFromPhrase error: %s", err)
+				canceler()
```

Comment:
yes, you can call `cancel` repeatedly.

https://github.com/golang/net/blob/master/context/pre_go17.go#L170

i'll remove conditional